app/utils/gtag.js

```diff
@@ -0,0 +1,5 @@
+export default function gtag() {
```

Comment:
So why doesn't this function take any arguments? It looks like you're calling it with arguments when you use it.

Reply:
I tried having the function take in arguments and then passed that object on to the dataLayer.  Unfortunately, it didn't seem to pass the event on to GA when I created the object, but worked fine when I passed along the Arguments object.  I wish I had a better explanation, but I don't.

Reply:
Can you point me to any documentation that you based this bit of code off of? I'm only asking because I was under the impression that the `<script>` tag you added to index.html loads a global function called `gtag` that you can call directly to track events. If that function exists then you might be overwriting here which could explain some of the weird behavior with the params.

Reply:
I can't point to any documentation I based this code off of.  I can't find any documentation on how to set up GA4 on Ember.

You are correct, the script added to index.html does add a global function called `gtag` that you can call directly to track events.  Unfortunately, without declaring gtag as a function within our code, none of our tests pass because we are calling an undeclared function.  We are definitely overwriting here, I just don't know of a better solution that both works and passes tests.

Reply:
Understood. I was referring to just documentation for gtag.js in general but I see that what you have up in index.html follows their docs. I'd like you to spend some more time trying to get the tests to pass without the need for this util. Having competing version of this function is likely to cause us headaches and we should be able to mock this in the tests somehow without having a whole function that only exists to make our tests happy.

Shot in the dark here but don't suppose you've tried adding your snippet in index.html to the index.html located in the `tests` folder?

Reply:
Adding the snippet to the tests/index.html did not appear to have any effect on the outcome of the tests.

I was able to get tests to pass without the additional function by excluding 'no-undef' from .eslintrc.js, and initial tests showed that it was tracking correctly in GA.  If you would prefer that approach, I can test and make sure all of the tracking is correct and then open a new pr.